### PR TITLE
Second signature registration transaction should update secondSignature flag for account - Closes #1147

### LINKED ISSUE #1147: Second signature registration transaction should update secondSignature flag for account
### Expected behavior
In the applyAsset function for Second signature transaction, the secondPublicKey and secondSignature property should be updated.
### Actual behavior
The applyAsset function only updates the secondPublicKey property.
### Which version(s) does this affect? (Environment, OS, etc...)
2.1.0

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

- secondSignature flag was not being set

### How did I fix it?

- secondSignature flag is being added on apply asset and removed in undo

### How to test it?

Build should be green

### Review checklist

* The PR resolves #1147 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-transactions/src/1_second_signature_transaction.ts [hunk 1]
```diff
@@ -209,6 +210,8 @@ export class SecondSignatureTransaction extends BaseTransaction {
 	protected undoAsset(store: StateStore): ReadonlyArray<TransactionError> {
 		const sender = store.account.get(this.senderId);
 		const { secondPublicKey, ...strippedSender } = sender;
```
Comment (shuse2) [inline packages/lisk-transactions/src/1_second_signature_transaction.ts]:
```suggestion
		const { secondPublicKey, secondSignature, ...strippedSender } = sender;
```

Comment (pablitovicente) [reply packages/lisk-transactions/src/1_second_signature_transaction.ts]: @shuse2 I think we need the properties to be present for stateStore to handle them correctly so instead changed to de-structuring as we discussed.

File: packages/lisk-transactions/src/1_second_signature_transaction.ts [hunk 2]
```diff
@@ -200,6 +200,7 @@ export class SecondSignatureTransaction extends BaseTransaction {
 		const updatedSender = {
 			...sender,
 			secondPublicKey: this.asset.signature.publicKey,
+			secondSignature: true,
```
Comment (SargeKhan) [inline packages/lisk-transactions/src/1_second_signature_transaction.ts]: It should be a number, with values either 0 or 1.

Comment (pablitovicente) [reply packages/lisk-transactions/src/1_second_signature_transaction.ts]: Done!

**Review by shuse2 — DISMISSED** [review]

File: packages/lisk-transactions/src/1_second_signature_transaction.ts [hunk 3]
```diff
@@ -208,7 +209,12 @@ export class SecondSignatureTransaction extends BaseTransaction {
 
 	protected undoAsset(store: StateStore): ReadonlyArray<TransactionError> {
 		const sender = store.account.get(this.senderId);
-		const { secondPublicKey, ...strippedSender } = sender;
+		const strippedSender = {
+			...sender,
+			secondPublicKey: undefined,
```
Comment (SargeKhan) [inline packages/lisk-transactions/src/1_second_signature_transaction.ts]: In Lisk-core, on undo action, the secondPublicKey property is assigned with the value `null`. Therefore, I think we should keep it consistent.

Comment (pablitovicente) [reply packages/lisk-transactions/src/1_second_signature_transaction.ts]: @SargeKhan  We have a TS lint rule that forbids null and favours undefined should I skip it in this case?

Comment (SargeKhan) [reply packages/lisk-transactions/src/1_second_signature_transaction.ts]: Yeah, I guess we have to do it in this particular case.

**Review by SargeKhan — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]

**Pull request merged**